packages/react-native-codegen/src/parsers/flow/modules/index.js

```diff
@@ -280,41 +300,32 @@ function translateTypeAnnotation(
                     ),
                   );
 
-                if (propertyTypeAnnotation.type === 'FunctionTypeAnnotation') {
-                  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
-                    hasteModuleName,
-                    property.value,
-                    property.key,
-                    propertyTypeAnnotation.type,
-                  );
-                }
-
-                if (propertyTypeAnnotation.type === 'VoidTypeAnnotation') {
-                  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
-                    hasteModuleName,
-                    property.value,
-                    property.key,
-                    'void',
-                  );
-                }
-
-                if (propertyTypeAnnotation.type === 'PromiseTypeAnnotation') {
-                  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
+                if (
+                  propertyTypeAnnotation.type === 'FunctionTypeAnnotation' ||
+                  propertyTypeAnnotation.type === 'PromiseTypeAnnotation' ||
+                  propertyTypeAnnotation.type === 'VoidTypeAnnotation'
+                ) {
+                  const invalidPropertyValueType =
+                    UnsupportedObjectPropertyTypeToInvalidPropertyValueTypeMap[
+                      propertyTypeAnnotation.type
+                    ];
+
+                  throwUnsupportedObjectPropertyValueTypeAnnotationParserError(
                     hasteModuleName,
                     property.value,
                     property.key,
-                    'Promise',
+                    invalidPropertyValueType,
                   );
+                } else {
```

Comment:
You don't need the `else`. If the function throws, the execution flow changes, so we can safely remove this indentation! ;)